src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -401,22 +401,24 @@ internal void SetTypeWithAnnotations(TypeWithAnnotations newType)
         /// Gets a new local symbol with the given TypeWithAnnotations as the new type. This
         /// type should be identical to the original except for nullability.
         /// </summary>
-        internal LocalSymbol WithAnalyzedType(TypeWithAnnotations analyzedType)
+        internal LocalSymbol WithContainingAndAnalyzedTypes(Symbol containingSymbol, TypeWithAnnotations analyzedType)
         {
+            Debug.Assert(SymbolEqualityComparer.IgnoreNullableAndTupleNames.Equals(ContainingSymbol, containingSymbol));
```

Comment:
>IgnoreNullableAndTupleNames [](start = 48, length = 27)

Is it a bug that tuple names may be distinct?

Reply:
Yes, [https://github.com/dotnet/roslyn/issues/35010](https://github.com/dotnet/roslyn/issues/35010)

---
In reply to: [327302307](https://github.com/dotnet/roslyn/pull/38769#discussion_r327302307) [](ancestors = 327302307)